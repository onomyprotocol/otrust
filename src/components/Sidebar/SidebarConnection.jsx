import React from 'react';
import styled from 'styled-components';

import { ConnectionStatus } from '../UI';
import { responsive } from 'theme/constants';

const Connection = styled.div`
  display: flex;
  flex-direction: column;
  gap: 20px;

  padding: 40px;

  @media screen and (max-width: ${responsive.laptop}) {
    padding: 24px;
  }

  @media screen and (max-width: ${responsive.tablet}) {
    flex-direction: row;
    gap: 40px;
  }

  @media screen and (max-width: ${responsive.smartphoneLarge}) {
    display: none;
  }
`;

const ConnectionRow = styled.div`
  display: flex;
  justify-content: space-between;

  @media screen and (max-width: ${responsive.tablet}) {
    flex-direction: column;
    justify-content: flex-start;
    gap: 10px;
  }

  > strong {
    font-weight: 400;
    color: ${props => props.theme.colors.textSecondary};
  }

  > span {
    font-weight: 500;
  }
`;

export default function SidebarConnection({ active, error, chainId, blockNumber }) {
  return (
    <Connection>
      <ConnectionRow>
        <strong>Connection</strong>
        <ConnectionStatus>{active ? 'Connected' : error ? 'Error' : 'Loading'}</ConnectionStatus>
      </ConnectionRow>
      <ConnectionRow>
        <strong>Chain Id</strong>
        <span>{chainId ?? ''}</span>
      </ConnectionRow>
      <ConnectionRow>
        <strong>Block Number</strong>
        <span>{blockNumber ? blockNumber.toFixed(0) : ''}</span>
      </ConnectionRow>
    </Connection>
  );
}
