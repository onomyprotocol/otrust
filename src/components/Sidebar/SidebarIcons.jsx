import React from 'react';

export function CloseIcon() {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 20 20" width="20" height="20">
      <g transform="translate(5 5.833)">
        <path
          d="M7.3.32a1.093 1.093 0 011.652 1.423l-.106.123-6.981 6.981A1.093 1.093 0 01.214 7.424L.32 7.3z"
          className="cls-2"
        ></path>
        <path
          d="M.32.32A1.092 1.092 0 011.742.214l.122.106L8.847 7.3a1.092 1.092 0 01-1.422 1.65L7.3 8.847.32 1.864A1.092 1.092 0 01.32.32z"
          className="cls-2"
        ></path>
      </g>
    </svg>
  );
}

export default CloseIcon;
