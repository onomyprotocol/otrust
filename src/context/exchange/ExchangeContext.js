import React, { createContext, useContext, useReducer, useState } from 'react';
import { BigNumber } from 'bignumber.js';

import { exchStringReducer, exchObjReducer } from 'context/exchange/ExchangeReducer';

export const ExchangeContext = createContext();
export const useExchange = () => useContext(ExchangeContext);

export const UpdateExchangeContext = createContext();
export const useUpdateExchange = () => useContext(UpdateExchangeContext);

function ExchangeProvider({ children }) {
  const [inputPending, setInputPending] = useState(false);

  const [objState, objDispatch] = useReducer(exchObjReducer, {
    askAmount: new BigNumber(0),
    bidAmount: new BigNumber(0),
    pendingTx: null,
    slippage: new BigNumber(0),
    txPending: false,
  });

  const [strState, strDispatch] = useReducer(exchStringReducer, {
    bidDenom: 'strong',
    input: '',
    output: '',
    status: '',
    strong: 'ETH',
    weak: 'wNOM',
  });

  const contextValue = {
    ...objState,
    ...strState,
    inputPending,
  };

  const updateValue = {
    objDispatch,
    strDispatch,
    setInputPending,
  };

  return (
    <UpdateExchangeContext.Provider value={updateValue}>
      <ExchangeContext.Provider value={contextValue}>{children}</ExchangeContext.Provider>
    </UpdateExchangeContext.Provider>
  );
}

export default ExchangeProvider;
