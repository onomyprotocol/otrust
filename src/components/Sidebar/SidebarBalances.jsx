import React from 'react';
import styled from 'styled-components';

import { PrimaryButton } from 'components/Modals/styles';
import { responsive } from 'theme/constants';
import { CloseIcon } from './SidebarIcons';

const Balances = styled.div`
  display: flex;
  flex-direction: column;
  gap: 22px;

  padding: 40px;

  border-bottom: 1px solid ${props => props.theme.colors.bgHighlightBorder};

  @media screen and (max-width: ${responsive.laptop}) {
    padding: 24px;
  }

  @media screen and (max-width: ${responsive.tablet}) {
    flex-direction: row;
    align-items: center;
    grid-column: 1/3;
    grid-row: 2/3;

    border: none;
  }

  @media screen and (max-width: ${responsive.tabletSmall}) {
    gap: 40px;
  }

  @media screen and (max-width: ${responsive.smartphoneLarge}) {
    padding: 20px;
  }

  strong {
    display: block;
    margin-bottom: 10px;

    font-weight: 400;
    color: ${props => props.theme.colors.textSecondary};
  }
`;

const BalanceNumber = styled.div`
  font-family: 'Bebas Neue', sans-serif;
  font-size: 28px;
  font-weight: ${props => (props.strong ? 700 : 400)};

  > small {
    margin-left: 5px;

    font-family: 'Poppins', sans-serif;
    color: ${props => props.theme.colors.textSecondary};
    font-size: 11px;
  }

  @media screen and (max-width: ${responsive.laptop}) {
    font-size: 20px;
  }
`;

const Balance = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
`;

const BalancePrice = styled.div``;
const BalanceHint = styled.div``;

const SecondaryIcon = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;

  height: 40px;
  width: 40px;

  background-color: ${props => props.theme.colors.bgHighlightBorder};
  border-radius: 8px;

  font-size: 20px;
  font-weight: 500;
  color: ${props => props.theme.colors.iconsSecondary};

  cursor: pointer;

  @media screen and (max-width: ${responsive.laptop}) {
    width: 32px;
    height: 32px;

    font-size: 14px;
  }

  @media screen and (max-width: ${responsive.tablet}) {
    margin-left: 24px;
  }

  &:hover {
    background-color: ${props => props.theme.colors.bgHighlightBorder_lighten};
  }

  &:active {
    background-color: ${props => props.theme.colors.bgHighlightBorder_darken};
  }
`;

const Approved = styled.div`
  display: inline-block;

  padding: 4px 30px 4px 8px;
  margin-top: 6px;

  position: relative;

  background-color: #2a303f;
  border-radius: 6px;

  color: ${props => props.theme.colors.highlightBlue};
  font-size: 10px;
  font-weight: 400;
  font-family: Poppins, sans-serif;

  @media screen and (max-width: ${responsive.tablet}) {
    margin: 0 0 0 12px;
  }

  svg {
    position: absolute;
    right: 4px;
    top: 2px;

    cursor: pointer;
    * {
      fill: currentColor;
    }
  }
`;

const WithdrawBtnWrapper = styled.div`
  margin-top: 20px;
  min-width: 150px;

  @media screen and (max-width: ${responsive.tablet}) {
    margin: 0 0 0 auto;
  }
`;

export default function SidebarBalances({ strong, weak, strongBalance, weakBalance }) {
  return (
    <Balances>
      <Balance>
        <BalancePrice>
          <strong>{strong} Balance</strong>
          <BalanceNumber>
            {strongBalance}
            <small> = $16,208.04</small>
          </BalanceNumber>
        </BalancePrice>
        <BalanceHint>
          <SecondaryIcon>i</SecondaryIcon>
        </BalanceHint>
      </Balance>
      <Balance>
        <BalancePrice>
          <strong>{weak} Balance</strong>
          <BalanceNumber strong>
            {weakBalance}
            <small> = $16,208.04</small>
            <Approved>
              <span>1,042 approved</span>
              <CloseIcon onClick={() => {}} />
            </Approved>
          </BalanceNumber>
        </BalancePrice>
        <BalanceHint>
          <SecondaryIcon>i</SecondaryIcon>
        </BalanceHint>
      </Balance>

      <WithdrawBtnWrapper>
        <PrimaryButton style={{ width: '100%' }}>Withdraw wNOM</PrimaryButton>
      </WithdrawBtnWrapper>
    </Balances>
  );
}
