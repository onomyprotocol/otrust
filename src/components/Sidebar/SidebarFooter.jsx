import React from 'react';
import styled from 'styled-components';

import { responsive } from 'theme/constants';
import { MediumIcon, TwitterIcon } from './SidebarIcons';

const SidebarFooterWrapper = styled.footer`
  display: flex;
  align-items: center;
  gap: 16px;

  padding: 50px 40px;
  margin-top: auto;

  @media screen and (max-width: ${responsive.laptop}) {
    padding: 40px 24px;
  }

  @media screen and (max-width: ${responsive.tablet}) {
    width: 100%;
    gap: 12px;

    position: absolute;
    bottom: 0;
    left: 0;
  }

  @media screen and (max-width: ${responsive.smartphoneLarge}) {
    padding: 24px 20px;

    background-color: ${props => props.theme.colors.bgDarken};
  }
`;

const Link = styled.a`
  display: block;
  margin-right: auto;

  color: ${props => props.theme.colors.textSecondary};
  text-decoration: none;

  &:hover {
    color: ${props => props.theme.colors.textSecondary_lighten};
  }

  &:active {
    color: ${props => props.theme.colors.textSecondary_darken};
  }
`;

const SecondaryIcon = styled.a`
  display: flex;
  align-items: center;
  justify-content: center;

  height: 40px;
  width: 40px;

  background-color: ${props => props.theme.colors.bgHighlightBorder};
  border-radius: 8px;

  font-size: 20px;
  color: ${props => props.theme.colors.iconsSecondary};

  cursor: pointer;

  svg {
    width: 20px;
    height: 20px;
  }

  @media screen and (max-width: ${responsive.laptop}) {
    width: 32px;
    height: 32px;

    font-size: 14px;

    svg {
      width: 16px;
      height: 16px;
    }
  }

  &:hover {
    background-color: ${props => props.theme.colors.bgHighlightBorder_lighten};
  }

  &:active {
    background-color: ${props => props.theme.colors.bgHighlightBorder_darken};
  }
`;

export default function SidebarFooter() {
  return (
    <SidebarFooterWrapper>
      <Link href="https://onomy.io/" target="_blank">
        About Onomy
      </Link>

      <SecondaryIcon href="https://medium.com/onomy-protocol" target="_blank">
        <MediumIcon />
      </SecondaryIcon>

      <SecondaryIcon href="https://twitter.com/onomyprotocol" target="_blank">
        <TwitterIcon />
      </SecondaryIcon>
    </SidebarFooterWrapper>
  );
}
