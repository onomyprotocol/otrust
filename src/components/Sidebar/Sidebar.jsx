import React from 'react';
import styled from 'styled-components';
import { useWeb3React } from '@web3-react/core';
import BigNumber from 'bignumber.js';

import { Panel } from 'components/UI';
import { useChain } from 'context/chain/ChainContext';
import { useExchange } from 'context/exchange/ExchangeContext';
import { responsive } from 'theme/constants';
import { format18 } from 'utils/math';
import SidebarHeader from './SidebarHeader';
import SidebarBalances from './SidebarBalances';
import SidebarFooter from './SidebarFooter';
import SidebarConnection from './SidebarConnection';

const SidebarLayout = styled.div`
  display: flex;
  flex-direction: column;

  height: 100%;

  @media screen and (max-width: ${responsive.tablet}) {
    display: grid;
    grid-template-columns: 1fr 300px;
    grid-template-rows: 100px 100px;
    justify-content: space-between;
  }

  @media screen and (max-width: ${responsive.tabletSmall}) {
    grid-template-columns: 1fr 250px;
  }

  @media screen and (max-width: ${responsive.smartphoneLarge}) {
    grid-template-columns: 100%;
    grid-template-rows: repeat(3, auto);
  }
`;

export default function AcctDash() {
  const { active, error, chainId, account } = useWeb3React();
  const { blockNumber, strongBalance, weakBalance } = useChain();
  const { strong, weak } = useExchange();

  return (
    <Panel>
      <SidebarLayout>
        <SidebarHeader account={account} />
        <SidebarBalances
          strong={strong}
          weak={weak}
          strongBalance={BigNumber.isBigNumber(strongBalance) ? `${format18(strongBalance).toFixed(6)}` : 'Loading'}
          weakBalance={BigNumber.isBigNumber(weakBalance) ? `${format18(weakBalance).toFixed(6)}` : 'Loading'}
        />
        <SidebarConnection active={active} error={error} chainId={chainId} blockNumber={blockNumber} />
        <SidebarFooter />
      </SidebarLayout>
    </Panel>
  );
}
