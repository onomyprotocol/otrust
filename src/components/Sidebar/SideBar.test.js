import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { BigNumber } from 'bignumber.js';

import { dark } from 'Theme/theme';
import { ChainContext } from '../../context/chain/ChainContext';
import { ExchangeContext } from '../../context/exchange/ExchangeContext';
import Sidebar from './Sidebar';

describe('Given the Sidebar component and strongBalance, weakBalance are of BigNumber type', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ThemeProvider theme={dark}>
          <ChainContext.Provider
            value={{
              blockNumber: 12345678,
              strongBalance: new BigNumber(0),
              weakBalance: new BigNumber(10),
            }}
          >
            <ExchangeContext.Provider value={{ strong: 'ETH', weak: 'wNOM' }}>
              <Sidebar />
            </ExchangeContext.Provider>
          </ChainContext.Provider>
        </ThemeProvider>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});

describe('Given the Sidebar component and strongBalance, weakBalance are NOT of BigNumber type', () => {
  describe('when the component is rendered', () => {
    afterEach(cleanup);

    it('should match the snapshot', () => {
      const { asFragment } = render(
        <ThemeProvider theme={dark}>
          <ChainContext.Provider
            value={{ blockNumber: 12345678, strongBalance: 0, weakBalance: 10, allowance: 'Loading' }}
          >
            <ExchangeContext.Provider value={{ strong: 'ETH', weak: 'wNOM' }}>
              <Sidebar />
            </ExchangeContext.Provider>
          </ChainContext.Provider>
        </ThemeProvider>,
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
