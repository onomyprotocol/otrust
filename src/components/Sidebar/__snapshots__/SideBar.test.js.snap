// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the Sidebar component and strongBalance, weakBalance are NOT of BigNumber type when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT bOaZEi"
  >
    <div
      class="sc-jffHpj dEbpYo"
    >
      <header
        class="sc-fujyAs gEMvSK"
      >
        <a
          class="sc-iCoGMd ftOiio"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h20v20H0z"
              fill="none"
            />
            <path
              d="M10.519 17.348H9.484a1.538 1.538 0 01-1.582-1.526 1.417 1.417 0 00-1.4-1.355 1.176 1.176 0 00-.669.172 1.61 1.61 0 01-.804.221 1.58 1.58 0 01-1.354-.774l-.525-.894a1.492 1.492 0 01-.021-1.524 1.569 1.569 0 01.608-.588 1.23 1.23 0 00.48-.476 1.327 1.327 0 00.141-1.016 1.367 1.367 0 00-.629-.845 1.522 1.522 0 01-.567-2.107l.509-.876a1.58 1.58 0 012.145-.565 1.41 1.41 0 00.668.167 1.458 1.458 0 001.245-.684 1.145 1.145 0 00.176-.655 1.331 1.331 0 01.21-.789 1.64 1.64 0 011.346-.774h1.068a1.6 1.6 0 011.359.774 1.326 1.326 0 01.205.789 1.15 1.15 0 00.175.655 1.462 1.462 0 001.245.68 1.433 1.433 0 00.673-.167 1.571 1.571 0 012.138.565l.509.878a1.509 1.509 0 01-.563 2.107 1.362 1.362 0 00-.635.847 1.318 1.318 0 00.148 1.014 1.169 1.169 0 00.472.476 1.717 1.717 0 01.616.588 1.512 1.512 0 01-.015 1.526l-.533.894a1.562 1.562 0 01-2.153.551 1.207 1.207 0 00-.669-.172 1.414 1.414 0 00-1.408 1.355 1.534 1.534 0 01-1.574 1.528zm-.51-9.543a2.136 2.136 0 00-2.161 2.107A2.057 2.057 0 008.48 11.4a2.181 2.181 0 001.528.611 2.1 2.1 0 001.521-3.591 2.142 2.142 0 00-1.52-.615z"
              fill="#8d89a3"
            />
          </svg>
        </a>
        <img
          alt=""
          class="sc-pNWdM jcyRRj"
          src="https://picsum.photos/72"
        />
        <a
          class="sc-iCoGMd ftOiio"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h20v20H0z"
              fill="none"
            />
            <path
              d="M18.643 10.674l-3.367 3.3a.987.987 0 01-1.4-1.392l1.137-1.129c.259-.257.173-.465-.192-.465H10.3a.984.984 0 110-1.968h4.525c.363 0 .451-.208.192-.466L13.88 7.425a.987.987 0 011.4-1.392l3.359 3.262a.957.957 0 01.004 1.379zM5.365 14.936h1.977a.987.987 0 110 1.974H4.379a1.317 1.317 0 01-1.314-1.316V4.41a1.317 1.317 0 011.314-1.316h2.963a.987.987 0 110 1.974H5.365a.327.327 0 00-.33.328v9.214a.329.329 0 00.33.326z"
              fill="#8d89a3"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <div
          class="sc-jrsJWt SXxEx"
        >
          <p>
            My Account
          </p>
          <span />
        </div>
      </header>
      <div
        class="sc-efHYUO hEBjDc"
      >
        <div
          class="sc-jNnpgg cJYNoe"
        >
          <div
            class="sc-dPaNzc"
          >
            <strong>
              ETH Balance
            </strong>
            <div
              class="sc-cTJkRt gsHzGO"
            >
              Loading
              <small>
                 = $16,208.04
              </small>
            </div>
          </div>
          <div
            class="sc-bBjRSN"
          >
            <a
              class="sc-cOifOu hA-dACZ"
            >
              i
            </a>
          </div>
        </div>
        <div
          class="sc-jNnpgg cJYNoe"
        >
          <div
            class="sc-dPaNzc"
          >
            <strong>
              wNOM Balance
            </strong>
            <div
              class="sc-cTJkRt fJbEIt"
            >
              Loading
              <small>
                 = $16,208.04
              </small>
              <div
                class="sc-Arkif dbmngf"
              >
                <span>
                  1,042 approved
                </span>
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    transform="translate(5 5.833)"
                  >
                    <path
                      class="cls-2"
                      d="M7.3.32a1.093 1.093 0 011.652 1.423l-.106.123-6.981 6.981A1.093 1.093 0 01.214 7.424L.32 7.3z"
                    />
                    <path
                      class="cls-2"
                      d="M.32.32A1.092 1.092 0 011.742.214l.122.106L8.847 7.3a1.092 1.092 0 01-1.422 1.65L7.3 8.847.32 1.864A1.092 1.092 0 01.32.32z"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
          <div
            class="sc-bBjRSN"
          >
            <a
              class="sc-cOifOu hA-dACZ"
            >
              i
            </a>
          </div>
        </div>
        <div
          class="sc-khIgEk cENgFj"
        >
          <button
            class="sc-dIsUp hPVUIx"
            style="width: 100%;"
          >
            Withdraw wNOM
          </button>
        </div>
      </div>
      <div
        class="sc-laZMeE bOwEdY"
      >
        <div
          class="sc-iNiQyp hoimnV"
        >
          <strong>
            Connection
          </strong>
          <div
            class="sc-gKAaRy iiHLvK"
          >
            Loading
          </div>
        </div>
        <div
          class="sc-iNiQyp hoimnV"
        >
          <strong>
            Chain Id
          </strong>
          <span />
        </div>
        <div
          class="sc-iNiQyp hoimnV"
        >
          <strong>
            Block Number
          </strong>
          <span>
            12345678
          </span>
        </div>
      </div>
      <footer
        class="sc-hTRkXV filpNh"
      >
        <a
          class="sc-jgPyTC egIKvM"
          href="https://onomy.io/"
          target="_blank"
        >
          About Onomy
        </a>
        <a
          class="sc-gSYDnn bLTUWz"
          href="https://medium.com/onomy-protocol"
          target="_blank"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              opacity="0.5"
            >
              <path
                d="M0 0h20v20H0z"
                fill="none"
              />
              <path
                d="M11.281 9.979A5.641 5.641 0 115.641 4.3a5.66 5.66 0 015.641 5.679m6.188 0c0 2.952-1.263 5.347-2.82 5.347s-2.82-2.394-2.82-5.347 1.261-5.346 2.818-5.346 2.82 2.394 2.82 5.347m2.531 0c0 2.645-.444 4.79-.992 4.79s-.992-2.145-.992-4.79.445-4.79.993-4.79S20 7.334 20 9.979"
                fill="#8d89a3"
              />
            </g>
          </svg>
        </a>
        <a
          class="sc-gSYDnn bLTUWz"
          href="https://twitter.com/onomyprotocol"
          target="_blank"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h20v20H0z"
              fill="none"
            />
            <path
              d="M16.555 6.438v.46a10.4 10.4 0 01-.66 3.6 10.744 10.744 0 01-1.961 3.282 9.729 9.729 0 01-3.2 2.4 10.445 10.445 0 01-4.383.921 10.247 10.247 0 01-2.922-.42 9.937 9.937 0 01-2.6-1.181q.42.041.86.04a7.14 7.14 0 002.381-.4 7.457 7.457 0 002.081-1.121 3.592 3.592 0 01-2.081-.73 3.463 3.463 0 01-1.261-1.771 3.579 3.579 0 00.66.06 3.7 3.7 0 00.941-.12 3.472 3.472 0 01-1.461-.67 3.567 3.567 0 01-1.031-1.261 3.749 3.749 0 01-.39-1.591v-.06a3.369 3.369 0 001.641.46 3.617 3.617 0 01-1.171-1.291 3.517 3.517 0 01-.43-1.711 3.574 3.574 0 01.48-1.8 10.262 10.262 0 007.4 3.762 4.156 4.156 0 01-.08-.82 3.5 3.5 0 01.48-1.8 3.627 3.627 0 011.311-1.311 3.462 3.462 0 011.791-.49 3.563 3.563 0 012.622 1.141 6.994 6.994 0 002.281-.881 3.481 3.481 0 01-.61 1.151 3.694 3.694 0 01-.971.851 7.316 7.316 0 002.081-.58 7.525 7.525 0 01-1.798 1.881z"
              fill="#8d89a3"
              opacity="0.5"
            />
          </svg>
        </a>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Given the Sidebar component and strongBalance, weakBalance are of BigNumber type when the component is rendered should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsrHT bOaZEi"
  >
    <div
      class="sc-jffHpj dEbpYo"
    >
      <header
        class="sc-fujyAs gEMvSK"
      >
        <a
          class="sc-iCoGMd ftOiio"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h20v20H0z"
              fill="none"
            />
            <path
              d="M10.519 17.348H9.484a1.538 1.538 0 01-1.582-1.526 1.417 1.417 0 00-1.4-1.355 1.176 1.176 0 00-.669.172 1.61 1.61 0 01-.804.221 1.58 1.58 0 01-1.354-.774l-.525-.894a1.492 1.492 0 01-.021-1.524 1.569 1.569 0 01.608-.588 1.23 1.23 0 00.48-.476 1.327 1.327 0 00.141-1.016 1.367 1.367 0 00-.629-.845 1.522 1.522 0 01-.567-2.107l.509-.876a1.58 1.58 0 012.145-.565 1.41 1.41 0 00.668.167 1.458 1.458 0 001.245-.684 1.145 1.145 0 00.176-.655 1.331 1.331 0 01.21-.789 1.64 1.64 0 011.346-.774h1.068a1.6 1.6 0 011.359.774 1.326 1.326 0 01.205.789 1.15 1.15 0 00.175.655 1.462 1.462 0 001.245.68 1.433 1.433 0 00.673-.167 1.571 1.571 0 012.138.565l.509.878a1.509 1.509 0 01-.563 2.107 1.362 1.362 0 00-.635.847 1.318 1.318 0 00.148 1.014 1.169 1.169 0 00.472.476 1.717 1.717 0 01.616.588 1.512 1.512 0 01-.015 1.526l-.533.894a1.562 1.562 0 01-2.153.551 1.207 1.207 0 00-.669-.172 1.414 1.414 0 00-1.408 1.355 1.534 1.534 0 01-1.574 1.528zm-.51-9.543a2.136 2.136 0 00-2.161 2.107A2.057 2.057 0 008.48 11.4a2.181 2.181 0 001.528.611 2.1 2.1 0 001.521-3.591 2.142 2.142 0 00-1.52-.615z"
              fill="#8d89a3"
            />
          </svg>
        </a>
        <img
          alt=""
          class="sc-pNWdM jcyRRj"
          src="https://picsum.photos/72"
        />
        <a
          class="sc-iCoGMd ftOiio"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h20v20H0z"
              fill="none"
            />
            <path
              d="M18.643 10.674l-3.367 3.3a.987.987 0 01-1.4-1.392l1.137-1.129c.259-.257.173-.465-.192-.465H10.3a.984.984 0 110-1.968h4.525c.363 0 .451-.208.192-.466L13.88 7.425a.987.987 0 011.4-1.392l3.359 3.262a.957.957 0 01.004 1.379zM5.365 14.936h1.977a.987.987 0 110 1.974H4.379a1.317 1.317 0 01-1.314-1.316V4.41a1.317 1.317 0 011.314-1.316h2.963a.987.987 0 110 1.974H5.365a.327.327 0 00-.33.328v9.214a.329.329 0 00.33.326z"
              fill="#8d89a3"
              fill-rule="evenodd"
            />
          </svg>
        </a>
        <div
          class="sc-jrsJWt SXxEx"
        >
          <p>
            My Account
          </p>
          <span />
        </div>
      </header>
      <div
        class="sc-efHYUO hEBjDc"
      >
        <div
          class="sc-jNnpgg cJYNoe"
        >
          <div
            class="sc-dPaNzc"
          >
            <strong>
              ETH Balance
            </strong>
            <div
              class="sc-cTJkRt gsHzGO"
            >
              0.000000
              <small>
                 = $16,208.04
              </small>
            </div>
          </div>
          <div
            class="sc-bBjRSN"
          >
            <a
              class="sc-cOifOu hA-dACZ"
            >
              i
            </a>
          </div>
        </div>
        <div
          class="sc-jNnpgg cJYNoe"
        >
          <div
            class="sc-dPaNzc"
          >
            <strong>
              wNOM Balance
            </strong>
            <div
              class="sc-cTJkRt fJbEIt"
            >
              0.000000
              <small>
                 = $16,208.04
              </small>
              <div
                class="sc-Arkif dbmngf"
              >
                <span>
                  1,042 approved
                </span>
                <svg
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    transform="translate(5 5.833)"
                  >
                    <path
                      class="cls-2"
                      d="M7.3.32a1.093 1.093 0 011.652 1.423l-.106.123-6.981 6.981A1.093 1.093 0 01.214 7.424L.32 7.3z"
                    />
                    <path
                      class="cls-2"
                      d="M.32.32A1.092 1.092 0 011.742.214l.122.106L8.847 7.3a1.092 1.092 0 01-1.422 1.65L7.3 8.847.32 1.864A1.092 1.092 0 01.32.32z"
                    />
                  </g>
                </svg>
              </div>
            </div>
          </div>
          <div
            class="sc-bBjRSN"
          >
            <a
              class="sc-cOifOu hA-dACZ"
            >
              i
            </a>
          </div>
        </div>
        <div
          class="sc-khIgEk cENgFj"
        >
          <button
            class="sc-dIsUp hPVUIx"
            style="width: 100%;"
          >
            Withdraw wNOM
          </button>
        </div>
      </div>
      <div
        class="sc-laZMeE bOwEdY"
      >
        <div
          class="sc-iNiQyp hoimnV"
        >
          <strong>
            Connection
          </strong>
          <div
            class="sc-gKAaRy iiHLvK"
          >
            Loading
          </div>
        </div>
        <div
          class="sc-iNiQyp hoimnV"
        >
          <strong>
            Chain Id
          </strong>
          <span />
        </div>
        <div
          class="sc-iNiQyp hoimnV"
        >
          <strong>
            Block Number
          </strong>
          <span>
            12345678
          </span>
        </div>
      </div>
      <footer
        class="sc-hTRkXV filpNh"
      >
        <a
          class="sc-jgPyTC egIKvM"
          href="https://onomy.io/"
          target="_blank"
        >
          About Onomy
        </a>
        <a
          class="sc-gSYDnn bLTUWz"
          href="https://medium.com/onomy-protocol"
          target="_blank"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              opacity="0.5"
            >
              <path
                d="M0 0h20v20H0z"
                fill="none"
              />
              <path
                d="M11.281 9.979A5.641 5.641 0 115.641 4.3a5.66 5.66 0 015.641 5.679m6.188 0c0 2.952-1.263 5.347-2.82 5.347s-2.82-2.394-2.82-5.347 1.261-5.346 2.818-5.346 2.82 2.394 2.82 5.347m2.531 0c0 2.645-.444 4.79-.992 4.79s-.992-2.145-.992-4.79.445-4.79.993-4.79S20 7.334 20 9.979"
                fill="#8d89a3"
              />
            </g>
          </svg>
        </a>
        <a
          class="sc-gSYDnn bLTUWz"
          href="https://twitter.com/onomyprotocol"
          target="_blank"
        >
          <svg
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h20v20H0z"
              fill="none"
            />
            <path
              d="M16.555 6.438v.46a10.4 10.4 0 01-.66 3.6 10.744 10.744 0 01-1.961 3.282 9.729 9.729 0 01-3.2 2.4 10.445 10.445 0 01-4.383.921 10.247 10.247 0 01-2.922-.42 9.937 9.937 0 01-2.6-1.181q.42.041.86.04a7.14 7.14 0 002.381-.4 7.457 7.457 0 002.081-1.121 3.592 3.592 0 01-2.081-.73 3.463 3.463 0 01-1.261-1.771 3.579 3.579 0 00.66.06 3.7 3.7 0 00.941-.12 3.472 3.472 0 01-1.461-.67 3.567 3.567 0 01-1.031-1.261 3.749 3.749 0 01-.39-1.591v-.06a3.369 3.369 0 001.641.46 3.617 3.617 0 01-1.171-1.291 3.517 3.517 0 01-.43-1.711 3.574 3.574 0 01.48-1.8 10.262 10.262 0 007.4 3.762 4.156 4.156 0 01-.08-.82 3.5 3.5 0 01.48-1.8 3.627 3.627 0 011.311-1.311 3.462 3.462 0 011.791-.49 3.563 3.563 0 012.622 1.141 6.994 6.994 0 002.281-.881 3.481 3.481 0 01-.61 1.151 3.694 3.694 0 01-.971.851 7.316 7.316 0 002.081-.58 7.525 7.525 0 01-1.798 1.881z"
              fill="#8d89a3"
              opacity="0.5"
            />
          </svg>
        </a>
      </footer>
    </div>
  </div>
</DocumentFragment>
`;
